util: do not use `mount-utils.IsLikelyNotMountPoint` anymore

nixpanic opened the pull request:
`IsLikelyNotMountPoint()` is an optimized version for `IsMountPoint()`
which can not detect all type of mounts (anymore). The slower
`IsMountPoint()` is more safe to use. This can cause a slight
performance regression in the case there are many mountpoints on the
system, but correctness is more important than speed while mounting.

Fixes: #4633

nixpanic committed:
util: do not use `mount-utils.IsLikelyNotMountPoint` anymore

`IsLikelyNotMountPoint()` is an optimized version for `IsMountPoint()`
which can not detect all type of mounts (anymore). The slower
`IsMountPoint()` is more safe to use. This can cause a slight
performance regression in the case there are many mountpoints on the
system, but correctness is more important than speed while mounting.

Fixes: #4633
Signed-off-by: Niels de Vos <ndevos@ibm.com>

iPraveenParihar commented:
/test ci/centos/mini-e2e-helm/k8s-1.32

Madhu-1 commented:
/test ci/centos/mini-e2e-helm/k8s-1.31

Madhu-1 commented:
@nixpanic we are using `IsLikelyNotMountPoint` in two other places (nfs,rbd) can we replace it as well?

nixpanic commented:
> @nixpanic we are using `IsLikelyNotMountPoint` in two other places (nfs,rbd) can we replace it as well?

Sure, done!

nixpanic commented:
/test ci/centos/mini-e2e-helm/k8s-1.32

Madhu-1 commented on internal/nfs/nodeserver/nodeserver.go:
```diff
@@ -200,7 +200,7 @@ func (ns *NodeServer) mountNFS(
 	)
 
-	notMnt, err := ns.Mounter.IsLikelyNotMountPoint(mountPoint)
+	notMnt, err := ns.Mounter.IsMountPoint(mountPoint)
```
we should change `notMnt` to `Ismnt` or something like that and we should reverse the check for it because `IsLikelyNotMountPoint` was returning its a mount point or not as `!` operation of IsMountPoint

nixpanic replied:
Ah, yes :facepalm:

nixpanic replied:
This should have been addressed now. Thanks for catching!

Madhu-1 commented on internal/nfs/nodeserver/nodeserver.go:
```diff
@@ -200,19 +200,19 @@ func (ns *NodeServer) mountNFS(
 				return err
 			}
-			notMnt = true
+			isMnt = false
```
i think this check is not required, because of an error, isMnt will be set to false by default

Madhu-1 commented on internal/rbd/nodeserver.go:
```diff
@@ -880,9 +880,9 @@ func (ns *NodeServer) mountVolume(ctx context.Context, stagingPath string, req *
 
 func (ns *NodeServer) createTargetMountPath(ctx context.Context, mountPath string, isBlock bool) (bool, error) {
```
can we update the `createTargetMountPath` to return isMnt to check it simple, returning `!` and doing `!` check on the function return value looks so confusing :(

nixpanic replied:
Sure, I can clean that up. It confused me a little too.

nixpanic replied:
@Madhu-1 , please have an other look. The 2nd commit simplifies `createTargetMountPath()`.

Madhu-1 reviewed (commented):
LGTM, if possible please see if you can address the nits.

nixpanic committed:
rbd: cleanup NodeServer.createTargetMountPath()

The inverse checking and returning of is-a-mounted-path makes it
difficult to understand the function. It is easier to follow the code
when the function just returns what it says it does, hence added the
comment for the function too.

Some errors were returned directly, others were converted to gRPC
errors. This has been corrected now too, and the caller converts the
plain error to a gRPC error now.

Signed-off-by: Niels de Vos <ndevos@ibm.com>

Madhu-1 reviewed (approved):
LGTM

iPraveenParihar commented:
/test ci/centos/mini-e2e-helm/k8s-1.32

Rakshith-R commented:
@mergifyio queue
